source/eventcore/drivers/posix/driver.d

```diff
@@ -116,12 +117,21 @@ final class PosixEventDriver(Loop : PosixEventLoop) : EventDriver {
 			return thname.length ? thname : "unknown";
 		}
 
-		if (m_loop.m_handleCount > 0) {
+		string leaking_handle_desc;
+		foreach (id, ref s; m_loop.m_fds){
+			if (!s.specific.hasType!(typeof(null)) && !(s.common.flags & FDFlags.internal) &&
+			   (!s.specific.hasType!(StreamSocketSlot) || s.streamSocket.state==ConnectionState.connected))
+					try{
+						if(s.specific.hasType!(StreamSocketSlot)){
+							print("state is: %d", s.streamSocket.state);
+						}
+						leaking_handle_desc ~= format!"   FD %s (%s)\n"(id, s.specific.kind);
+					} catch(Exception ex) { print("exception happened in Driver.dispose() during formatting"); }
+		}
+
+		if(leaking_handle_desc!=""){
 			print("Warning (thread: %s): leaking eventcore driver because there are still active handles", getThreadName());
-			foreach (id, ref s; m_loop.m_fds)
-				if (!s.specific.hasType!(typeof(null)) && !(s.common.flags & FDFlags.internal))
-					print("   FD %s (%s)", id, s.specific.kind);
-			return false;
```

Comment:
was there any reason why we returned from here? there are some cleanup operation after this statement

Reply:
Probably because those cleanups are not safe if there are leaking handles ?